frontend/src/app/restaurant/page.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+
+interface Restaurant {
+  name: string
+  address: string
+  cuisine: string
+  rating: string
+  phone: string
+  image_url: string
+  website_url: string
+}
+
+interface RestaurantProps {
+  restaurant: Restaurant
+}
+
+interface GetRestaurantsResponse {
+  data: Restaurant[]
+}
+
+const RestaurantCard: React.FC<RestaurantProps> = ({ restaurant }) => {
+  return (
+    <>
+      <h1>{restaurant.name}</h1>
+      <div>{restaurant.address}</div>
+      <div>{restaurant.cuisine}</div>
+      <div>{restaurant.rating}</div>
+      <div>{restaurant.phone}</div>
+      <div>{restaurant.image_url}</div>
+      <div>{restaurant.website_url}</div>
+      <br />
+    </>
+  )
+}
+
+const getRestaurants = async (): Promise<Restaurant[]> => {
```

Comment:
I suggest creating a special class that will have a responsibility for the communication with the backend and move `getRestaurants` there. `HttpClient`? `Router`?

Reply:
Resolving this comment as the updated code is using fetch directly with useEffect right now. We can definitely consider moving it out later! I just want to close this issue so people have things to work off of.